From 75014d5e2c0b2528833ed6d9edcfc1c113589c18 Mon Sep 17 00:00:00 2001
From: Giuseppe CAVALLARO <peppe.cavallaro@st.com>
Date: Wed, 15 Oct 2008 11:01:37 +0200
Subject: [PATCH] stmmac: stx7141 GMII clock configuration

Provide the right clock configuration to support the GMII mode as well.

Signed-off-by: Giuseppe Cavallaro <peppe.cavallaro@st.com>
Signed-off-by: Stuart Menefy <stuart.menefy@st.com>
---
 arch/sh/kernel/cpu/sh4/setup-stx7141.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/arch/sh/kernel/cpu/sh4/setup-stx7141.c b/arch/sh/kernel/cpu/sh4/setup-stx7141.c
index 16828e0..138b4b3 100644
--- a/arch/sh/kernel/cpu/sh4/setup-stx7141.c
+++ b/arch/sh/kernel/cpu/sh4/setup-stx7141.c
@@ -672,6 +672,15 @@ void stx7141_configure_ethernet(int port, int reverse_mii, int mode,
 	sc = sysconf_claim(SYS_CFG, 7, 16+port, 16+port, "stmmac");
 	sysconf_write(sc, 1);
 
+	/* GMII clock configuration */
+	if (port == 0) {
+		sc = sysconf_claim(SYS_CFG, 7, 13, 13, "stmmac");
+		sysconf_write(sc, 1);
+	} else if (port == 1) {
+		sc = sysconf_claim(SYS_CFG, 7, 15, 15, "stmmac");
+		sysconf_write(sc, 1);
+	}
+
 	/* enmii: Interface type (rev MII/MII) */
 	sc = sysconf_claim(SYS_CFG, 7, port ? 31 : 27, port ? 31 : 27,
 			   "stmmac");
-- 
1.5.3.6

